File: Travel Package Management Java/src/com/example/travel/TravelPackageTest.java
package com.example.travel;

import org.junit.Test;
import static org.junit.Assert.*;

public class TravelPackageTest {
    
    @Test
    public void testAddDestination() {
        TravelPackage adventureTrip = new TravelPackage("Adventure Trip", 10);
        Destination himalayas = new Destination("Himalayas");
        adventureTrip.addDestination(himalayas);

        assertEquals(1, adventureTrip.getItinerary().size());
        assertEquals("Himalayas", adventureTrip.getItinerary().get(0).getName());
    }

    @Test
    public void testAddPassenger() {
        TravelPackage adventureTrip = new TravelPackage("Adventure Trip", 10);
        Passenger amaan = new StandardPassenger("Amaan", 2, 300);
        adventureTrip.addPassenger(amaan);

        assertEquals(1, adventureTrip.getPassengerList().size());
        assertEquals("Amaan", adventureTrip.getPassengerList().get(0).getName());
    }
}

File: Travel Package management C++/TravelPackageTest.cpp
#include "TravelPackage.h"
#include "Destination.h"
#include "Passenger.h"
#include "StandardPassenger.h"
#include "Activity.h"
#include <gtest/gtest.h>
#include <memory>

TEST(TravelPackageTest, AddDestination) {
    auto travelPackage = std::make_shared<TravelPackage>("Adventure Trip", 10);
    auto destination = std::make_shared<Destination>("Himalayas");
    travelPackage->addDestination(destination);

    auto itinerary = travelPackage->getItinerary();

    EXPECT_EQ(itinerary.size(), 1);

    EXPECT_EQ(itinerary.front()->getName(), "Himalayas");
}


TEST(TravelPackageTest, AddPassenger_Success) {
    auto travelPackage = std::make_shared<TravelPackage>("Adventure Trip", 10);
    auto passenger = std::make_shared<StandardPassenger>("Amaan", 2, 300);
    travelPackage->addPassenger(passenger);

    auto passengerList = travelPackage->getPassengerList();

    EXPECT_EQ(passengerList.size(), 1);

    EXPECT_EQ(passengerList.front()->getName(), "Amaan");
}
